blog/_posts/2025-08-14-node-autoprovisioning.md

```diff
@@ -0,0 +1,76 @@
+
+Node auto provisioning takes the requirements set by the user in the workload deployment file, and custom resource definition (CRD) files such as the [NodePool](https://learn.microsoft.com/azure/aks/node-autoprovision-node-pools) and [AKSNodeClass](https://learn.microsoft.com/azure/aks/node-autoprovision-aksnodeclass), and provisions nodes that will meet these criteria.
+
+![nap-how-it-works](/assets/images/aks-nap/nap-how-it-works-image.png)
+
+NAP works at the infrastructure level, and adjusts the quantity and sizes of VMs. NAP can be paired with application-level scalers, which affect CPU/Memory resource allocation and pod replica count, such as [KEDA](https://learn.microsoft.com/azure/aks/keda-about), [Horizontal Pod Autoscaler](https://learn.microsoft.com/azure/aks/concepts-scale#horizontal-pod-autoscaler), or [Vertical Pod Autoscaler](https://learn.microsoft.com/en-us/azure/aks/vertical-pod-autoscaler). For more on this, check out our [AKS Scalers Deep Dive on Youtube](https://www.youtube.com/watch?v=oILHg5hsZQ0).
+
+## Node auto provisioning vs. cluster autoscaler
+
+Cluster autoscaler, the standard Kubernetes autoscaler solution, requires the use of same VM size node pools, and scales pre-existing node pools up or down. NAP works instead at the cluster level, and manages single-instance VMs, also handling the provisioning experience for multiple VM sizes and architecture at once. NAP allows for better bin-packing, cost savings, and performance than cluster autoscaler.
+
+## NAP vs Self-Hosted Karpenter
+
+Karpenter is the OSS project that schedules workloads for efficient compute usage. Our [AKS Karpenter Provider (self-hosted)](https://github.com/Azure/karpenter-provider-azure) makes use of Karpenter on Azure available. Node Auto-provisioning (NAP) is our managed add-on for Karpenter on AKS that manages certain aspects of the Karpenter experience on Azure. NAP is the recommended mode for most users for a few reasons.
+
+NAP manages:
+
+- Node Image upgrades (Linux)
+- Kubernetes version upgrades
+- Karpenter version updates
+- VM OS disk updates
+- Karpenter Logs (through Azure Monitor)
+- Metrics (through Managed Prometheus)
+
+In self-hosted Karpenter, users are responsible for managing these processes. Self-hosted mode is useful for advanced users who want to customize or experiment with Karpenter's deployment. The managed add-on NAP simplifies this experience and allows you to focus on your workloads rather than infrastructure.
+
+## Getting Started with NAP
+
+To get started with NAP, you can visit the [Node Auto Provisioning documentation](https://learn.microsoft.com/azure/aks/node-autoprovision). The documentation includes resources, requirements, and all the info needed to enable node auto provisioning in your cluster today.
+
+## Roadmap + Next Steps
+
+We’re continuing to expand the capabilities for NAP for additional feature support and performance. Our upcoming roadmap of feature support includes:
+
```

Comment:
Provide a little bit more color what are doing first , milestones tentatively , GH roadmap links to these ?